src/tiled/automapper.cpp

```diff
@@ -749,7 +797,7 @@ static bool compareLayerTo(const TileLayer *setLayer,
                 if (listNo.isEmpty()) {
                     if (matchListYes)
                         continue;
-                    if (!ruleDefinedListYes && !cells.contains(c1))
```

Comment:
Maybe we should add some other layer types (on top of regions, inputyes, inputnot) to handle such cases. Leaving all listYes empty is sometimes necessary when we want to create a region rule where not all input tiles are adjacent(like an X shaped one ), so I think it is better if an empty input tile has the expected result, and use some additional explicit layers for other cases.